stitch/inspect.go

```
@@ -0,0 +1,99 @@
+package stitch
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"text/scanner"
+)
+
+const quiltPath = "QUILT_PATH"
+
+type argOption struct {
+	result interface{} // Optionally store result.
+}
+
+func usage() {
+	fmt.Fprintln(os.Stderr, "Usage: inspect <path to spec file> [commands]\n"+
+		"Options\n"+
+		" - viz <pdf|ascii>\n"+
+		"Dependencies\n"+
+		" - easy-graph (install Graph::Easy from cpan)\n"+
+		" - graphviz (install from your favorite package manager)\n")
+	os.Exit(1)
+}
+
+func initializeGraph(spec Stitch) (graph, error) {
+	g := makeGraph()
+
+	for label := range spec.QueryLabels() {
+		g.addNode(label)
+	}
+	g.addNode(PublicInternetLabel)
+
+	for _, conn := range spec.QueryConnections() {
+		err := g.addConnection(conn.From, conn.To)
+		if err != nil {
+			return graph{}, err
+		}
+	}
+
+	for _, pl := range spec.QueryPlacements() {
+		err := g.addPlacementRule(pl)
+		if err != nil {
+			return graph{}, err
+		}
+	}
+
+	for _, m := range spec.QueryMachines() {
+		g.machines = append(g.machines, getMachine(m))
+	}
+
+	return g, nil
+}
+
+// InspectMain is the main function for inspect tool. Helps visualize stitches.
+func InspectMain(opts []string) {
+	configPath := opts[1]
+
+	f, err := os.Open(configPath)
+	if err != nil {
+		panic(err)
```

I'm not sure if a panic message and stacktrace is appropriate for mistyping a file name. Maybe just throw an error instead?

```
aetherboard:quilt shwangster$ quilt inspect a
panic: open a: no such file or directory

goroutine 1 [running]:
panic(0x9c8120, 0xc820210150)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/NetSys/quilt/stitch.InspectMain(0xc8200621c0, 0x2, 0x2)
        /Users/shwangster/go/src/github.com/NetSys/quilt/stitch/inspect.go:62 +0xaf
main.main()
        /Users/shwangster/go/src/github.com/NetSys/quilt/quilt.go:63 +0x59e
```